pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -73,12 +73,20 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 	}
 
 	spp := dv.oc.Properties.ServicePrincipalProfile
-	token, err := aad.GetToken(ctx, dv.log, spp.ClientID, string(spp.ClientSecret), dv.subscriptionDoc.Subscription.Properties.TenantID, dv.env.Environment().ActiveDirectoryEndpoint, dv.env.Environment().ResourceManagerEndpoint)
+
+	spToken, err := aad.NewServicePrincipalToken(spp.ClientID, string(spp.ClientSecret), dv.subscriptionDoc.Subscription.Properties.TenantID, dv.env.Environment().ActiveDirectoryEndpoint, dv.env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return err
+	}
+
+	tokenAuthorizer := refreshable.NewAuthorizer(spToken)
+
+	err = aad.AuthenticateServicePrincipalToken(ctx, dv.log, tokenAuthorizer, nil)
 	if err != nil {
 		return err
 	}
 
-	spAuthorizer := refreshable.NewAuthorizer(token)
+	spAuthorizer := refreshable.NewAuthorizer(spToken)
```

Comment:
What is the difference between `spAuthorizer` and `tokenAuthorizer`? can we replace the use of `spAuthorizer` with the new `tokenAuthorizer`?